WooCommerce/Classes/Authentication/Store Creation/Plan/WebCheckoutViewModel.swift

```diff
@@ -0,0 +1,69 @@
+import WebKit
+
+/// View model used for the web view controller to check out.
+final class WebCheckoutViewModel: AuthenticatedWebViewModel {
+    // `AuthenticatedWebViewModel` protocol conformance.
+    let title = Localization.title
+    let initialURL: URL?
+
+    /// Keeps track of whether the completion has been triggered so that it's only invoked once.
+    /// There are usually multiple redirects with the same success URL prefix.
+    private var isComplete: Bool = false
+
+    private let completion: () -> Void
+
+    /// - Parameters:
+    ///   - siteSlug: The slug of the site URL that the web checkout is for.
+    ///   - completion: Invoked when the webview reaches the success URL.
+    init(siteSlug: String, completion: @escaping () -> Void) {
+        self.initialURL = URL(string: String(format: Constants.checkoutURLFormat, siteSlug))
+        self.completion = completion
+    }
+
+    func handleDismissal() {
+        // no-op: dismissal is handled in the close button in the navigation bar.
```

Comment:
This method is to add any tracks needed when the view is dismissed, not about implementing the dismissal 😅 Sorry if the name is misleading.